packages/components/scripts/wrapper-generator/VueWrapperGenerator.ts

```diff
@@ -0,0 +1,144 @@
+
+export class VueWrapperGenerator extends AbstractWrapperGenerator {
+  protected packageDir = 'components-vue';
+  protected projectDir = 'vue-wrapper';
+
+  public getComponentFileName(component: TagName): string {
+    return `${pascalCase(component.replace('p-', ''))}Wrapper.vue`;
+  }
+
+  public generateImports(component: TagName, extendedProps: ExtendedProp[], nonPrimitiveTypes: string[]): string {
+    const hasEventProps = extendedProps.some(({ isEvent }) => isEvent);
+    const hasProps = getComponentMeta(component).props;
+
+    const vueImports = ['onMounted', 'onUpdated', 'ref'];
+    const importsFromVue = hasProps ? `import { ${vueImports.join(', ')} } from 'vue';` : '';
+
+    const utilsImports = [
+      'getPrefixedTagName',
+      ...(hasProps ? ['syncProperties'] : []),
+      ...(hasEventProps ? ['addEventListenerToElementRef'] : []),
+    ];
+    const importsFromUtils = `import { ${utilsImports.join(', ')} } from '../../utils';`;
+
+    const importsFromTypes = nonPrimitiveTypes.length
+      ? `import type { ${nonPrimitiveTypes.join(', ')} } from '../types';`
+      : '';
+
+    return `<script setup lang="ts">
+  ${[importsFromVue, importsFromUtils, importsFromTypes].filter((x) => x).join('\n  ')}`;
+  }
+
+  public generateProps(component: TagName, rawComponentInterface: string): string {
+    const propsName = this.generatePropsName(component);
+
+    return getComponentMeta(component).props
+      ? `  type ${propsName} = {${rawComponentInterface
+          .slice(1, -1)
+          .split(';\n')
+          .filter((x) => !x.match(/ {2}on[A-Z][a-z]+.+/))
+          .join(';\n')}  }`
+      : '';
+  }
+
+  public generateComponent(component: TagName, extendedProps: ExtendedProp[]): string {
+    const propsName = this.generatePropsName(component);
+    const eventNamesAndTypes = extendedProps
+      .filter(({ isEvent }) => isEvent)
+      .map(({ key, rawValueType }) => {
+        const [, type] = /<(\w+)>/.exec(rawValueType) ?? [];
+        return {
+          eventName: camelCase(key.replace('on', '')),
+          type,
+        };
+      });
+
+    const disableEslintForLine = ' // eslint-disable-line vue/require-valid-default-prop';
+
+    const defaultPropsWithValue = extendedProps
+      .map(({ key, defaultValue, isEvent }) => {
+        // TODO: better approach to identify object?
+        if (!(isEvent || defaultValue === undefined)) {
+          const transformedDefaultValue = defaultValue.startsWith('{') ? `() => (${defaultValue})` : defaultValue;
+          return `${key}: ${transformedDefaultValue},${
+            component === 'p-headline' && key === 'color' ? disableEslintForLine : ''
+          }`;
+        } else {
+          return undefined;
+        }
+      })
+      .filter((x) => x)
+      .join('\n     ');
+
+    const componentMeta = getComponentMeta(component);
+    const hasEvent = componentMeta.hasEvent;
+    const hasProps = componentMeta.props;
+    const hasSlot = componentMeta.hasSlot;
```

Comment:
Avoid potential aliasing by destructuring.
```suggestion
    const { hasEvent, hasProps, hasSlot } = getComponentMeta(component);
```